lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -370,14 +371,15 @@ def present_strategy(compute_api, wished_server):
     else:
         target_server = query_results[0]
 
-    if server_attributes_should_be_changed(compute_api=compute_api, target_server=target_server,
-                                           wished_server=wished_server):
+    if server_or_ip_should_be_changed(compute_api=compute_api, target_server=target_server,
+                                      wished_server=wished_server):
         changed = True
 
         if compute_api.module.check_mode:
             return changed, {"status": "Server %s attributes would be changed." % target_server["id"]}
 
         target_server = server_change_attributes(compute_api=compute_api, target_server=target_server, wished_server=wished_server)
+        target_server = server_change_ip_attribute(compute_api=compute_api, target_server=target_server, wished_server=wished_server)
```

Comment:
Both `server_change_attributes` and `server_change_ip_attribute` methods call `compute_api.get(path="servers/%s" % target_server["id"])`. If only mutable attributes but no IP are updated (or only IP but no other mutable attributes), `compute_api.get` is called twice whereas one call would be sufficient. Then `server_change_ip_attribute` could be called by `server_change_attributes`, before calling `compute_api.get`.